app/javascript/hooks/useLazyFetch.ts

```diff
@@ -104,13 +86,29 @@ const useLazyFetchCore = <T>(
   };
 };
 
-export const useLazyFetch = <T>(initialData: T, options: UseLazyFetchOptions<T>): UseLazyFetchResult<T> =>
-  useLazyFetchCore(initialData, options, (hasLoaded) => !hasLoaded);
+const useFetchOnMount = (options: { fetchOnMount?: boolean }, hasLoaded: boolean, fetchFn: () => Promise<void>) => {
+  const fetchOnMount = options.fetchOnMount ?? true;
```

Comment:
`fetchOnMount` doesn't sound right to me either; on mount sounds like it only triggers when the component first renders, but actually it will fetch on a later render if the value changes to true - maybe `fetchUnlessLoaded` (for want of a better name)?